synapse/http/replicationagent.py

```diff
@@ -58,15 +61,25 @@ def endpointForURI(self, uri: URI) -> IStreamClientEndpoint:
 
         Returns: The correct client endpoint object
         """
-        if uri.scheme in (b"http", b"https"):
-            endpoint = HostnameEndpoint(self.reactor, uri.host, uri.port)
-            if uri.scheme == b"https":
+        # The place to connect to now comes in as the name of the worker, similar to
+        # a hostname in placement. Use the instance_map data to get the actual
+        # connection information.
+        worker_name = uri.netloc.decode("utf-8")
+        netloc = self.instance_map[worker_name].netloc()
+        scheme = self.instance_map[worker_name].scheme()
+
+        if scheme in ("http", "https"):
+            host, port = netloc.split(":", maxsplit=1)
+            endpoint = HostnameEndpoint(self.reactor, host, int(port))
```

Comment:
```suggestion
            endpoint = HostnameEndpoint(self.reactor, self.instance_map[worker_name].host, self.instance_map[worker_name].port)
```

Reply:
(And similarly below.)

Reply:
Applied suggestion in [cdc5bc2](https://github.com/matrix-org/synapse/pull/15578/commits/cdc5bc286619ed121f4d39f48dcc97e44e8cf325) and a little bit of fixup in [36f5339](https://github.com/matrix-org/synapse/pull/15578/commits/36f5339a8f088b79560ff52dca47a3d50071b29a)